lib/src/main/java/io/ably/lib/http/TokenAuth.java

```diff
@@ -36,13 +36,13 @@ public void setTokenDetails(TokenDetails tokenDetails) {
 		this.encodedToken = Base64Coder.encodeString(tokenDetails.token).replace("=", "");
 	}
 
-	public TokenDetails authorise(AuthOptions options, TokenParams params) throws AblyException {
+	public boolean authorise(AuthOptions options, TokenParams params, boolean force) throws AblyException {
```

Comment:
In an earlier version of these changes I was calling it from elsewhere with `force` explicitly `true`, irrespective of `options.force`, but that later disappeared.

This is also called from `ensureValidAuth()` (https://github.com/ably/ably-java/blob/0a0f3ecb1d0810d26fceb4e2f9b80614649ed642/lib/src/main/java/io/ably/lib/rest/Auth.java#L383) with an explicit `false`, and it is possible for `Auth.authOptions` to have `force=true` because that isn't prohibited in the `authOptions` passed in the constructor. I guess [this](https://github.com/ably/ably-java/blob/0a0f3ecb1d0810d26fceb4e2f9b80614649ed642/lib/src/main/java/io/ably/lib/rest/Auth.java#L732) should have a `.storedValues()` as well.